examples/servo/src/servo_util.rs

```diff
@@ -11,85 +12,107 @@ use euclid::{Scale, Size2D};
 use i_slint_core::items::ColorScheme;
 use slint::{ComponentHandle, SharedString};
 
-use servo::{
-    RenderingContext, Servo, ServoBuilder, Theme, WebViewBuilder, webrender_api::units::DevicePixel,
-};
+use servo::{Servo, ServoBuilder, Theme, WebViewBuilder, webrender_api::units::DevicePixel};
 
 use crate::{
-    Palette, WebviewLogic,
+    MyApp, Palette, WebviewLogic,
     adapter::{SlintServoAdapter, upgrade_adapter},
     delegate::AppDelegate,
+    on_events::on_app_callbacks,
     rendering_context::ServoRenderingAdapter,
     waker::Waker,
 };
 
-#[cfg(not(target_os = "android"))]
-use crate::rendering_context::try_create_gpu_context;
-
-#[cfg(target_os = "android")]
-use crate::rendering_context::create_software_context;
+pub fn init_servo(
+    app: MyApp,
+    initial_url: SharedString,
+    #[cfg(not(target_os = "android"))] device: slint::wgpu_27::wgpu::Device,
+    #[cfg(not(target_os = "android"))] queue: slint::wgpu_27::wgpu::Queue,
+) -> Rc<SlintServoAdapter> {
+    let app_weak = app.as_weak();
+    let (waker_sender, waker_receiver) = channel::unbounded::<()>();
+
+    #[cfg(not(target_os = "android"))]
+    let adapter = Rc::new(SlintServoAdapter::new(
+        app_weak,
+        waker_sender.clone(),
+        waker_receiver.clone(),
+        device,
+        queue,
+    ));
+
+    #[cfg(target_os = "android")]
+    let adapter = Rc::new(SlintServoAdapter::new(
+        app_weak,
+        waker_sender.clone(),
+        waker_receiver.clone(),
+    ));
 
-pub fn spin_servo_event_loop(state: Rc<SlintServoAdapter>) {
-    let state_weak = Rc::downgrade(&state);
+    let state_weak = Rc::downgrade(&adapter);
 
     slint::spawn_local({
```

Comment:
The block spawns here is a future but there is no call to await. Why is spawn_local needed?